olp-cpp-sdk-dataservice-read/src/PrefetchTilesProvider.h

```diff
@@ -81,6 +81,7 @@ class PrefetchTilesProvider final {
   std::shared_ptr<repository::DataRepository> dataRepo_;
   std::shared_ptr<repository::PrefetchTilesRepository> prefetchTilesRepo_;
   std::shared_ptr<olp::client::OlpClientSettings> settings_;
+  std::string layer_id_;
```

Comment:
```suggestion
  const std::string layer_id_;
```
Consider to making the member immutable if it doesn't change in the implementation. The reason is to protect the impl code from random change of this variable.